test/adapter/json_api/pagination_links_test.rb

```diff
@@ -21,30 +21,70 @@ def setup
           ]
         end
 
-        def mock_request(query_parameters = {}, original_url = URI)
+        def test_pagination_links_using_kaminari
+          assert_links_serialization(expected_response_with_pagination_links, kaminari_collection, options)
+        end
+
+        def test_pagination_links_using_will_paginate
+          assert_links_serialization(expected_response_with_pagination_links, will_paginate_collection, options)
+        end
+
+        def test_pagination_links_with_additional_params
+          assert_links_serialization(expected_response_with_pagination_links_and_additional_params, will_paginate_collection, options(test: 'test'))
+        end
+
+        def test_pagination_config_set_to_false
+          previous_config = ActiveModelSerializers.config.collection_serializer
+          ActiveModelSerializers.config.collection_serializer = ActiveModel::Serializer::NonPaginatedCollectionSerializer
+          assert_links_serialization(paginated_data, kaminari_collection, options)
+        ensure
+          ActiveModelSerializers.config.collection_serializer = previous_config
+        end
+
+        def test_not_showing_pagination_links
+          assert_links_serialization(non_paginated_data, @array)
+        end
+
+        def test_non_paginated_serializer
+          assert_links_serialization(paginated_data, kaminari_collection, options.merge(serializer: ActiveModel::Serializer::NonPaginatedCollectionSerializer))
+        end
+
+        def test_last_page_pagination_links_using_kaminari
+          assert_links_serialization(expected_response_with_last_page_pagination_links, kaminari_collection(3), options)
+        end
+
+        def test_last_page_pagination_links_using_will_paginate
+          assert_links_serialization(expected_response_with_last_page_pagination_links, will_paginate_collection(3), options)
+        end
+
+        private
+
+        def options(query_parameters = {}, original_url = URI)
```

Comment:
Well the function is called `mock_request` but build a mocked SerializationContext and sets the `@options`. I think this hide even more information and make it more complex to infer from the name. Also, the fact that `mock_request` sets the `@options` before the adapter was created messed up some tests and I go fed up with its behavior. I don't mind using name containing `mock_request` but I'd rather have it returns the options instead of setting the instance variable. Would `options_with_mocked_request` work for you or does it still hide what's going on?